liiatools/common/aggregate.py

```diff
@@ -21,28 +22,29 @@ def __init__(self, fs: FS, config: PipelineConfig):
         self.fs = fs
         self.config = config
 
-    def list_directories(self) -> List[str]:
+    def list_files(self) -> List[str]:
         """
-        List the directories in the current directory.
+        List the files in the current directory.
         """
         return sorted(self.fs.listdir("/"))
 
     def current(self) -> DataContainer:
         """
         Get the current session as a datacontainer.
         """
-        directories = self.list_directories()
-        return self.combine_files(directories)
+        files = self.list_files()
+        return self.combine_files(files)
 
-    def load_file(self, directory) -> DataContainer:
+    def load_file(self, file) -> DataContainer:
         """
         Load a file from the current directory.
         """
-        current_dir = self.fs.opendir(directory)
         data = DataContainer()
+        table_id = re.search(r"_([a-zA-Z0-9]*)\.", file)
+
         for table_spec in self.config.table_list:
-            if current_dir.exists("ssda903_" + table_spec.id + ".csv"):
-                with current_dir.open("ssda903_" + table_spec.id + ".csv", "rb") as f:
+            if table_id and table_id.group(1) == table_spec.id:
```

Comment:
It might be worth adding brackets here as such `table_id and (table_id.group(1) == table_spec.id)` .
There is a risk that it might be run as `(table_id and table_id.group(1)) == table_spec.id`, if not explicitly specified. 

If we're sure that the == will always be run before the 'and' then this (without the brackets) is fine.

Reply:
Thanks Tambe, from my understanding Python will always run the `if table_id` first which is equivalent to `if bool(table_id)` just to check that it does exist. If `table_id` does not exist then the second part of the if statement will produce an error as there will be no `table_id.group(1)`. That's why I have done it this way, to catch the error at a good time. I will do some further testing though as I might be wrong

Reply:
Hi Patrick,
If that's the default way of running it, then this works.
All good.